### ISSUE TITLE
need item rename blacklist

### ISSUE THREAD

**Issue by ProtoGhost** [Zrips/CMI#6719]
Hi, I'd like to request you to add a feature where we can blacklist certain names in itemrename. We use plugins that make use of itemnames for its property. so it seems like players with rename access can easily exploit it. So can u please add the said feature? will be really helpful. Thanks!

**Comment by ProtoGhost** [comment]
ItemRenaming: # When set to true, players will be denyied from renaming defined items in anvil or with itemname command # Option to define specific name by using regex format # Can be bypassed with cmi.anvil.itemrename.bypass Prevent: true # When set to true item renaming will be disabled in general, with anvil or command independent of renamed material # This can be bypassed with previous permission node 
GlobalDisable: false # List of materials followed with optional regex code which can prevent specific naming, like renaming mob spawners into another type while still allowing renaming spawner into anything else 
List: - mobspawner:([A-z]+ (?i)\w*spawner) - ender_chest - chest
names-blacklist:
          names: [] #names can be added as string to either blacklist those names from being named or only whitelist those names
           whitelist: #when set to true all above given strings will be added as a whitelist so items can only be renamed to those given strings.. When set to false given strings will act as a blacklist 

MAIN USE - Helps to deny people from naming explicit names to items

**Comment by mrfloris** [comment]
Why can't you use the existing blacklist?

**Comment by CloudeLecaw** [comment]
Its allready possible to blacklist items with its name there
you just need to enter the right regex.

**Comment by Kiran-GC** [comment]
Hi , i'm an admin from the issue creator's server ( @ProtoGhost ) . The main reason we asked for the said feature is to block some existing exploits on our server. Yes i am aware that i can blacklist names for specific materials. But the issue is that one of the plugins we use makes use of ItemName to give the said powers of the item ignoring the material. Yes ik its the dumbest thing I've ever seen. So in short the plugin only checks for itemnames and not the material. which means that anyone can rename the item and get the effect of the item.(I hope this explained what  i meant.) We already did report this issue to the plugin dev but i also asked for a feature here just cause i thought it will be good to have a blacklisting feature for names since many players tend to add explicit names for items. Hope this cleared every doubts.

**Comment by CloudeLecaw** [comment]
outch..
The thing your plugin does is really one of the worst solutions i ever heared of.
But ok in this case i can unterstand why you wanee block just an itemname to give instead of renaming a choosen item

Welll still no nice behave of that plugin. your might really should ask the dev to add a material scan too.
be able to rename any item just to get effects can be dangerouse..

Well on our server we used Colored Items names for this and forbit our players renaming with colors.

**Comment by Kiran-GC** [comment]
Yes , that's much better. @CloudeLecaw . And yes at first i thought that the plugin we used already had material scan so i had blacklisted the material when we developed the server. But turns out the plugin only checked for itemname 😂 . Anyways the dev have released a build with material scan after our request. but yeh adding a feature like this will help if issues like this arise in the future

**Comment by Kiran-GC** [comment]
This feature was never added.....

**Comment by mrfloris** [comment]
Correct, suggestions are posted and when zrips has the time to review something he will read through the discussion(s) and consider implementation of it. Posting a suggestion doesn't guarantee something being added. 

I also want to point out that despite me and others contributing to a discussion, that it's just our opinion or sharing of information. We don't speak for zrips, we can all say yes, and he can say no, and visa versa. 

Also, this is posted under Issue, whereas feature enhancements and suggestions should be posted under discussions. 

Maybe @Zrips can start by moving it to the discussion section?

**Comment by CloudeLecaw** [comment]
i never said its added @Kiran-GC this is also not what suggestions are for.
Suggestion is like mrfloris said to give some ideas to zrips and if he has time he cansider about it or not.

**Comment by CloudeLecaw** [comment]
Pls move it to discussions :)
